src/user-safe/UserSafe.sol

```diff
@@ -0,0 +1,678 @@
+        uint256 minUsdcAmount,
+        uint256 amountUsdcToSend,
+        bytes calldata swapData
+    ) external onlyEtherFiCashSafe {
+        if (
+            amountWeETHToSwap + blockedFundsForWithdrawal[weETH] >
+            IERC20(weETH).balanceOf(address(this))
+        ) revert InsufficientBalance();
+
+        uint256 returnAmount = _swapWeETHToUsdc(
+            amountWeETHToSwap,
+            minUsdcAmount,
+            swapData
+        );
+        if (amountUsdcToSend > returnAmount)
+            revert AmountGreaterThanUsdcReceived();
+
+        _checkSpendingLimit(usdc, amountUsdcToSend);
+        IERC20(usdc).safeTransfer(msg.sender, amountUsdcToSend);
+
+        emit SwapTransferForSpending(amountWeETHToSwap, amountUsdcToSend);
+    }
+
+    function _getSpendingLimitRenewalTimestamp(
+        uint64 startTimestamp,
+        SpendingLimitTypes spendingLimitType
+    ) internal pure returns (uint64 renewalTimestamp) {
+        if (spendingLimitType == SpendingLimitTypes.Daily)
+            return startTimestamp + 24 * 60 * 60;
+        else if (spendingLimitType == SpendingLimitTypes.Weekly)
+            return startTimestamp + 7 * 24 * 60 * 60;
+        else if (spendingLimitType == SpendingLimitTypes.Monthly)
+            return startTimestamp + 30 * 24 * 60 * 60;
+        else return startTimestamp + 365 * 24 * 60 * 60;
+    }
+
+    function _swapWeETHToUsdc(
+        uint256 amount,
+        uint256 minUsdcAmount,
+        bytes calldata swapData
+    ) internal returns (uint256) {
+        IERC20(weETH).safeTransfer(address(_swapper), amount);
+        return _swapper.swap(weETH, usdc, amount, minUsdcAmount, swapData);
+    }
+
+    function _setSpendingLimit(
+        uint8 spendingLimitType,
+        uint256 limitInUsd
+    ) internal {
+        _spendingLimit = SpendingLimitData({
+            spendingLimitType: SpendingLimitTypes(spendingLimitType),
+            renewalTimestamp: _getSpendingLimitRenewalTimestamp(
+                uint64(block.timestamp),
+                SpendingLimitTypes(spendingLimitType)
+            ),
+            spendingLimit: limitInUsd,
+            usedUpAmount: 0
+        });
+
+        emit SetSpendingLimit(spendingLimitType, limitInUsd);
+    }
+
+    function _setIncomingSpendingLimit(
+        uint8 spendingLimitType,
+        uint256 limitInUsd
+    ) internal {
+        _incomingSpendingLimit = SpendingLimitData({
+            spendingLimitType: SpendingLimitTypes(spendingLimitType),
+            renewalTimestamp: _getSpendingLimitRenewalTimestamp(
+                _spendingLimit.renewalTimestamp,
+                SpendingLimitTypes(spendingLimitType)
+            ),
+            spendingLimit: limitInUsd,
+            usedUpAmount: 0
+        });
+
+        emit SetIncomingSpendingLimit(spendingLimitType, limitInUsd);
+    }
+
+    function _requestWithdrawal(
+        address[] memory tokens,
+        uint256[] memory amounts,
+        address recipient
+    ) internal {
```

Comment:
so it will override if there was any already pending withdrawal request?

Reply:
in that case, we will see 

some emitted `WithdrawalRequested`,
but never emit `WithdrawalProcessed`

Reply:
Yes but that depends on the user, they would ideally want to wait instead of just request to pull out their funds. I think this can be managed. Do you think we should implement some other way?

Reply:
I am fine with the current simple case.

but we can check if there is any pending one and cancelt it emitting `WithdrawalCanceled` event

Reply:
Ohh yes we can do that, we can cancel previous withdrawal and add a new one. I can add a new event there. Will add.

Reply:
Added this feature where a `WithdrawalCancelled` event is emitted when a withdrawal request is overridden

Reply:
thx